kernel-rs/src/fs/lfs/inode.rs

```diff
@@ -208,8 +203,42 @@ impl InodeGuard<'_, Lfs> {
         // * dip is inside bp.data.
         // * dip will not be read.
 
-        // TODO: use transaction to write
-        // tx.write(bp, ctx);
+        let dip = unsafe {
+            &mut *(bp.deref_inner_mut().data.as_mut_ptr() as *mut Dinode)
+                .add(self.inum as usize % IPB)
+        };
+
+        let inner = self.deref_inner();
+        match inner.typ {
+            InodeType::Device { major, minor } => {
+                dip.typ = DInodeType::Device;
+                dip.major = major;
+                dip.minor = minor;
+            }
+            InodeType::None => {
+                dip.typ = DInodeType::None;
+                dip.major = 0;
+                dip.minor = 0;
+            }
+            InodeType::Dir => {
+                dip.typ = DInodeType::Dir;
+                dip.major = 0;
+                dip.minor = 0;
+            }
+            InodeType::File => {
+                dip.typ = DInodeType::File;
+                dip.major = 0;
+                dip.minor = 0;
+            }
+        }
+
+        (*dip).nlink = inner.nlink;
+        (*dip).size = inner.size;
+        for (d, s) in (*dip).addr_direct.iter_mut().zip(&inner.addr_direct) {
+            *d = *s;
+        }
+        (*dip).addr_indirect = inner.addr_indirect;
+        tx.write(bp, ctx);
```

Comment:
I think that the `InodeGuard::update` method, as implemented here, is used to commit one inode update to disk. However, in LFS, shouldn't the updated inode be buffered in the segment buffer first before being flushed to the disk.